guidance/library/_capture.py

```diff
@@ -0,0 +1,12 @@
+from .._guidance import guidance
+from .._grammar import capture as grammar_capture, GrammarFunction
+
+
+@guidance(stateless=lambda *args, **kwargs: isinstance(args[0], GrammarFunction))
+def capture(lm, value, name):
+    if isinstance(value, GrammarFunction):
+        return lm + grammar_capture(value, name)
+    else:
+        start_len = len(lm)
+        lm += value
+        return lm.set(name, str(lm)[start_len:])
```

Comment:
Just brainstorming... I _think_ running str(lm)[start_len:] after lm += value should always work, but there's no chance of any placeholders left behind in the str(lm) representation by the time the capture executes, right?

Reply:
I think it is safe, or at least the best we can do. People need to be able to use the captured value right away, so whatever the lm str rep is at that point is what they get (which won't be f-strings since those are grammars not models).